extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -50,64 +59,90 @@ export const HistoryItem = ({
     .split(" ")
     .slice(1, 3)
     .join(" ");
-  const operationAssetCode = assetCode || "XLM";
+  const srcAssetCode = sourceAssetCode || "XLM";
+  const destAssetCode = assetCode || "XLM";
 
   let isRecipient = false;
   let paymentDifference = "";
+  let rowText = "";
+  let dateText = "";
   let IconComponent = <Icon.Shuffle className="HistoryItem__icon--default" />;
   let PaymentComponent = null as React.ReactElement | null;
 
-  if (isPayment) {
+  let transactionDetailProps: TransactionDetailProps = {
+    operation,
+    isRecipient,
+    isPayment,
+    isSwap,
+    headerTitle: "",
+    operationText: "",
+    externalUrl: `${url}/op/${id}`,
+    setIsDetailViewShowing,
+  };
+
+  if (isSwap) {
+    PaymentComponent = (
+      <>
+        {new BigNumber(amount).toFixed(2, 1)} {destAssetCode}
+      </>
+    );
+    rowText = t(`${srcAssetCode} for ${destAssetCode}`);
```

Comment:
ah, this type of string interpolation won't work with the `t()` function unfortunately.
It'll need to be: 
```
headerTitle: t(`Swapped {{srcAssetCode}} for {{destAssetCode}}`, {
        srcAssetCode,
        destAssetCode,
      }),
```

https://www.i18next.com/translation-function/interpolation#basic